book/src/pong-tutorial/pong-tutorial-05.md

```diff
@@ -0,0 +1,887 @@
+its direction and put it back in the middle of the screen.
+
+Now, we just need to add our new system to `main.rs`, and we should be able to
+keep playing after someone scores and log who got the point.
+
+```rust,no_run,noplaypen
+# extern crate amethyst;
+#
+# use amethyst::{
+#    prelude::*,
+#    core::transform::TransformBundle,
+#    renderer::{
+#        DisplayConfig,
+#        DrawSprite,
+#        Pipeline,
+#        RenderBundle,
+#        Stage,
+#    }
+# };
+#
+# mod systems {
+#     use amethyst;
+#     pub struct PaddleSystem;
+#     impl<'a> amethyst::ecs::System<'a> for PaddleSystem {
+#         type SystemData = ();
+#         fn run(&mut self, _: Self::SystemData) { }
+#     }
+#     pub struct MoveBallsSystem;
+#     impl<'a> amethyst::ecs::System<'a> for MoveBallsSystem {
+#         type SystemData = ();
+#         fn run(&mut self, _: Self::SystemData) { }
+#     }
+#     pub struct BounceSystem;
+#     impl<'a> amethyst::ecs::System<'a> for BounceSystem {
+#         type SystemData = ();
+#         fn run(&mut self, _: Self::SystemData) { }
+#     }
+#     pub struct WinnerSystem;
+#     impl<'a> amethyst::ecs::System<'a> for WinnerSystem {
+#         type SystemData = ();
+#         fn run(&mut self, _: Self::SystemData) { }
+#     }
+# }
+#
+# fn main() -> amethyst::Result<()> {
+#
+# let path = "./resources/display_config.ron";
+# let config = DisplayConfig::load(&path);
+# let pipe = Pipeline::build().with_stage(Stage::with_backbuffer()
+#       .clear_target([0.0, 0.0, 0.0, 1.0], 1.0)
+#       .with_pass(DrawSprite::new()),
+# );
+# let input_bundle = amethyst::input::InputBundle::<String, String>::new();
+#
+let game_data = GameDataBuilder::default()
+#    .with_bundle(RenderBundle::new(pipe, Some(config)).with_sprite_sheet_processor())?
+#    .with_bundle(TransformBundle::new())?
+#    .with_bundle(input_bundle)?
+#    .with(systems::PaddleSystem, "paddle_system", &["input_system"])
+#    .with(systems::MoveBallsSystem, "ball_system", &[])
+#    .with(
+#        systems::BounceSystem,
+#        "collision_system",
+#        &["paddle_system", "ball_system"],
+#    )
+    // --snip--
+    .with(systems::WinnerSystem, "winner_system", &["ball_system"]);
+#
+# Ok(())
+# }
+```
+
+
+## Adding a Scoreboard
+
+We have a pretty functional Pong game now! At this point, the least fun thing
+about the game is just that players have to keep track of the score themselves.
+Our game should be able to do that for us.
+
+In this section, we'll set up UI rendering for our game and create a scoreboard
+to display our players scores.
+
+First, let's add the UI rendering in `main.rs`:
+
+```rust,no_run,noplaypen
+# extern crate amethyst;
+#
+use amethyst::{
+#     prelude::*,
+#     core::transform::TransformBundle,
+#     renderer::{
+#         DisplayConfig,
+#         DrawSprite,
+#         Pipeline,
+#         RenderBundle,
+#         Stage,
+#     },
+    // --snip--
```

Comment:
What's that for?

Reply:
It's just to show that some things have been hidden from the example (past code). That's what the Rust book does, so I thought it would work here. Let me know if it's not very clear though.

Reply:
Thanks, makes sense.